games/static/games/proto.js

```diff
@@ -327,7 +356,7 @@ class Board {
       // Sometimes if the interval is just right an alien gets picked up
       // instead of the rocket, the following if statement takes care of
       // this situation by setting the square back to rocket if it is an alien
-      if (rocketInFlight.isAlien()) {
+      if (rocketInFlight && rocketInFlight.isAlien()) {
```

Comment:
mmm how can it be that you get it null if you pick it up from `rocketLocation` ?

Reply:
it has to do with the interval, sometimes you shoot the alien so the rocket goes but the empty square gets picked up so this takes care of this, if its not there i get console errors sometimes

Reply:
returning nulls is a bad practice because you never know if its an error, or not found or anything.
and its not clear from reading this code as well
so one option is to leave a comment but like we said before- it means we lost - we couldnt express it via code so we had to leave a comment.
and also it wouldnt solve the problem that the next developer who wants to add a feature will use the function `this.liftRocketAt(rocketLocation)` without knowing it could return null.
so, i would suggest having a function that checks if rocket is at its expected location.
i see you get it from here:
`let rocketLocation = this.getIndexOf(rocket);`
i'm not sure where rocket index is kept (i would expect it to be on rocket).
if so, i belive that function should be a method on rocket - `rocket.isReallyAtLocation(index)` or something
if not, it can be a method on `board`

Reply:
btw - if you feel its too much for this PR, just add a comment and open a task on the board